spice/spice.py

```diff
@@ -230,6 +231,9 @@ def __init__(
         logging_dir: Optional[Path | str] = None,
         logging_callback: Optional[Callable[[SpiceResponse, str, str], None]] = None,
         default_temperature: Optional[float] = None,
+        retry_count: int = 0,
+        retry_wait: float = 1,
+        retry_backoff: float = 1.5,
```

Comment:
It feels kind of weird having this be in the constructor. I think I would prefer just having a good default and always using that. I'm also not entirely sure this is necessary because I think the openai library uses backoff (although I could be wrong on that, it was a while ago).